### PR TITLE
WIP accept language header

### PR DESCRIPTION
**Pull request by JackQuincy**
**What this PR does / why we need it**:
This adds accept language  header on calls to ARM
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Allows us to get localized error messages from the services we call
**Special notes for your reviewer**:
This is how their team suggested we add these headers here https://github.com/Azure/azure-sdk-for-go/issues/658 Looking at it. It is a little cumbersome and makes me think we should push harder to get native support in the actual sdk. I'll point them here
**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/808)

### CODE REVIEW

File: pkg/armhelpers/compute.go [hunk 1]
```diff
@@ -2,24 +2,116 @@ package armhelpers
...
 import (
 	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/go-autorest/autorest"
 )
 
 // ListVirtualMachines returns (the first page of) the machines in the specified resource group.
-func (az *AzureClient) ListVirtualMachines(resourceGroup string) (compute.VirtualMachineListResult, error) {
-	return az.virtualMachinesClient.List(resourceGroup)
+func (az *AzureClient) ListVirtualMachines(resourceGroup string) (result compute.VirtualMachineListResult, err error) {
+	client := az.virtualMachinesClient
```
Comment (anhowe) [inline pkg/armhelpers/compute.go]: Put a comment on each of these calls, to say replace with the one liner, once they added the accept language header.

Comment (JackQuincy) [inline pkg/armhelpers/compute.go]: This PR was more to serve as a case in point that we needed something better. There is discussion in the issue on the SDK and they provided a better way we can use to do this with much less code.

**Comment by weinong** [comment]
Reviewed 3 of 9 files at r1, 1 of 6 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[cmd/root.go, line 63 at r2](https://reviewable.io:443/reviews/azure/acs-engine/808#-Kn1gKhlhcjCSpEKyvkw:-Kn1gKhlhcjCSpEKyvkx:b-h6cg2w) ([raw file](https://github.com/azure/acs-engine/blob/2074a96a564779e3625182205e9172bde8dd17b5/cmd/root.go#L63)):*
> ```Go
> 	CertificatePath string
> 	PrivateKeyPath  string
> 	language        string
> ```

why is it private?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/azure/acs-engine/808)*

**Comment by JackQuincy** [comment]
*[cmd/root.go, line 63 at r2](https://reviewable.io:443/reviews/azure/acs-engine/808#-Kn1gKhlhcjCSpEKyvkw:-Kn1iXdBx6h1tE6TLDPk:bhkv38t) ([raw file](https://github.com/azure/acs-engine/blob/2074a96a564779e3625182205e9172bde8dd17b5/cmd/root.go#L63)):*
<details><summary><i>Previously, weinong (Weinong Wang) wrote…</i></summary><blockquote>

why is it private?
</blockquote></details>

mostly because I didn't need it anywhere outside of this file. I could export it if you feel strongly

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/azure/acs-engine/808)*

**Comment by weinong** [comment]
Reviewed 1 of 6 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[cmd/root.go, line 63 at r2](https://reviewable.io:443/reviews/azure/acs-engine/808#-Kn1gKhlhcjCSpEKyvkw:-Kn1jN-0CWHbdk9W7g68:bflv7fj) ([raw file](https://github.com/azure/acs-engine/blob/2074a96a564779e3625182205e9172bde8dd17b5/cmd/root.go#L63)):*
<details><summary><i>Previously, JackQuincy (Jack) wrote…</i></summary><blockquote>

mostly because I didn't need it anywhere outside of this file. I could export it if you feel strongly
</blockquote></details>

i dont mind either way.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/azure/acs-engine/808)*

**Pull request merged**